src/api.rs

```diff
@@ -167,6 +173,7 @@ impl Config {
     match key {
       "low_latency" => self.enc.low_latency = value.parse().map_err(|_e| ParseError)?,
       "key_frame_interval" => self.enc.key_frame_interval = value.parse().map_err(|_e| ParseError)?,
+      "max_key_frame_interval" => self.enc.max_key_frame_interval = value.parse().map_err(|_e| ParseError)?,
```

Comment:
This API is opposite of x264, which is kind of confusing. I'd rather "key frame interval" be the max, and having a separate min option. In particular, most users shouldn't need to adjust the minimum interval - in fact, if our detector is smart enough we should be able to set it to 1. (it probably isn't right now)

Reply:
Yeah, I chose to have a minimum in order to account for scenes that might have rapid motion and would falsely detect multiple consecutive keyframes, even though it would be less efficient. Xvid avoids the need for this by adjusting the threshold to be less likely to pick a keyframe if it found one recently, and more likely if it's close to the max distance. (Despite its age, Xvid's scenecut detection is pretty good.) Ours could do something similar in the future.

For now I can flip the settings to be more intuitive.